packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -821,13 +821,18 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
 
         tryCancel();
 
+        // A request can have some request-specific timeouts defined. We check if they have been defined, and if so,
+        // use those timeouts instead. However; if they aren't present, the defaults will be used.
+        const requestTimeoutMillis = request?.requestTimeoutMillis ?? this.internalTimeoutMillis;
+        const requestHandlerTimeoutMillis = request?.requestHandlerTimeoutMillis ?? this.requestHandlerTimeoutMillis;
```

Comment:
Why not

```js
            request.requestHandlerTimeoutSecs ??= this.requestHandlerTimeoutMillis / 1000;
            [...]
            await this._timeoutAndRetry(
                () => source.markRequestHandled(request!),
                request.requestHandlerTimeoutSecs * 1000,
                `Marking request ${request.url} (${request.id}) as handled timed out after ${requestTimeoutMillis / 1e3} seconds.`,
            );
```

Reply:
Ah yes true, nullish coalescing reassignment. I rarely use it, but this is definitely a good use-case